librad/src/net/peer.rs

```diff
@@ -0,0 +1,195 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(dead_code)]
+
+use std::fmt;
+
+use serde::{
+    de::{MapAccess, Visitor},
+    Deserialize,
+    Deserializer,
+    Serialize,
+    Serializer,
+};
+
+use crate::{
+    hash::Hash,
+    keys::device,
+    paths::Paths,
+    uri::{self, RadUrn},
+};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Rev {
+    Git(git2::Oid),
+}
+
+impl Serialize for Rev {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Self::Git(oid) => {
+                let mut seq = vec![0];
```

Comment:
Could you leave a comment saying that this is a tag for the variant? Just to make sure it's called out and not something "magic" :)

Reply:
Yes, but no. This is actually the standard CBOR encoding for a tuple, so instead of writing a letter to future self whenever we have to convince `serde` to output proper CBOR, we should rather fix the abstraction leak (hint, hint #102)

Reply:
Heh, fair enough :smile: